### DISCUSSION TITLE
Windows Package Manager 1.5.441-preview

### DISCUSSION THREAD

**Discussion by ryfu-msft** [microsoft/winget-cli#2960] [Releases]
This is the second development build after the Windows Package Manager 1.4 build for Windows 10 (1809+) and Windows 11. This build will be released to Windows Insider Dev builds and Windows Package Manager Insiders.

Experimental features are enabled in this release. The experimental feature for package pinning is now supported and included in this release. 

Run `winget features` to see which experimental features are enabled or disabled.
Add the following to your settings (`winget settings`) file to enable the experimental features including package pinning:

```JSON
    "experimentalFeatures": {
	  "pinning": true,
	  "dependencies": true,
	  "directMSI": true,
	  "uninstallPreviousArgument": true,
    },
```

This release includes an early preview of our Microsoft.WinGet.Client PowerShell module. Improvements to the PowerShell module have been made in this release to enhance the output of the cmdlets. Information about getting started and usage can be found [here](https://github.com/microsoft/winget-cli/tree/master/src/PowerShell/Microsoft.WinGet.Client#quick-start-guide).

## Features
* Pin a package #476 by @florelis 
* Open log files or provide path as output #2355 by @Trenly
* Show which admin setting has been enabled/disabled in confirmation string #2846 by @Trenly 

## What's Changed
* Make --Open-Logs Stable by @Trenly in https://github.com/microsoft/winget-cli/pull/2841
* Add --custom argument for passing additional installer arguments by @Trenly in https://github.com/microsoft/winget-cli/pull/2832
* Add database for tracking pins and base implementation for pin commands by @florelis in https://github.com/microsoft/winget-cli/pull/2769
* Show which admin setting has been enabled/disabled in confirmation string by @Trenly in https://github.com/microsoft/winget-cli/pull/2846
* Refactor some code into shared library by @JohnMcPMS in https://github.com/microsoft/winget-cli/pull/2844
* Add rest interface 1.4 to supported list by @yao-msft in https://github.com/microsoft/winget-cli/pull/2853
* Block msix provisioning api calls where known OS bugs exist by @yao-msft in https://github.com/microsoft/winget-cli/pull/2855
* Allow Version Listing through 'Winget Search' by @Trenly in https://github.com/microsoft/winget-cli/pull/2847
* Refactor arg validation  by @florelis in https://github.com/microsoft/winget-cli/pull/2862
* Rename privacy.md to PRIVACY.md by @WilliamDavidHarrison in https://github.com/microsoft/winget-cli/pull/2907
* fix: remove extra space by @WilliamDavidHarrison in https://github.com/microsoft/winget-cli/pull/2904
* chore: remove blank line by @WilliamDavidHarrison in https://github.com/microsoft/winget-cli/pull/2906
* feat: update pr template by @WilliamDavidHarrison in https://github.com/microsoft/winget-cli/pull/2905
* feat(template): update title + desc for feature request by @WilliamDavidHarrison in https://github.com/microsoft/winget-cli/pull/2915
* feat(template): update title + desc for docs report by @WilliamDavidHarrison in https://github.com/microsoft/winget-cli/pull/2914
* feat(template): update title + desc for bug report by @WilliamDavidHarrison in https://github.com/microsoft/winget-cli/pull/2913
* Remove use of Invoke-Expression in test script by @florelis in https://github.com/microsoft/winget-cli/pull/2921
* Show Enabled Admin Settings in --info by @Trenly in https://github.com/microsoft/winget-cli/pull/2901
* Update zlib library in Pure project by @ryfu-msft in https://github.com/microsoft/winget-cli/pull/2923
* Fix GetFullNameFromFamilyName for non-elevated context by @yao-msft in https://github.com/microsoft/winget-cli/pull/2922
* Allow multiple apps in a single command by @florelis in https://github.com/microsoft/winget-cli/pull/2861
* Use C# wrapper objects for PowerShell cmdlet output by @ryfu-msft in https://github.com/microsoft/winget-cli/pull/2871
* Fix behavior for user settings scope preference/requirement for portable install by @ryfu-msft in https://github.com/microsoft/winget-cli/pull/2918
* Fix Summary telemetry event by @yao-msft in https://github.com/microsoft/winget-cli/pull/2941
* Implement package pinning by @florelis in https://github.com/microsoft/winget-cli/pull/2813
* Spec for package pinning by @yao-msft in https://github.com/microsoft/winget-cli/pull/2611
* Fix spelling from pinning spec by @yao-msft in https://github.com/microsoft/winget-cli/pull/2946
* Give admin access to temp folder by @yao-msft in https://github.com/microsoft/winget-cli/pull/2945
* Down sampling telemetry events by @yao-msft in https://github.com/microsoft/winget-cli/pull/2950
* Add support for elevation requirement in COM by @ryfu-msft in https://github.com/microsoft/winget-cli/pull/2919

## New Contributors
* @WilliamDavidHarrison made their first contribution in https://github.com/microsoft/winget-cli/pull/2907

**Full Changelog**: https://github.com/microsoft/winget-cli/compare/v1.5.101-preview...v1.5.441-preview

<hr /><em>This discussion was created from the release <a href='https://github.com/microsoft/winget-cli/releases/tag/v1.5.441-preview'>Windows Package Manager 1.5.441-preview</a>.</em>

**Comment by fates7777** [comment]
how install?

**Reply by Trenly** [reply]
Head over to the [Releases](https://github.com/microsoft/winget-cli/releases) page, go to [1.5.441-preview](https://github.com/microsoft/winget-cli/releases/tag/v1.5.441-preview), download the MSIXBUNDLE, and then open the file. It should prompt to install automatically.

Alternatively, run `Add-AppxPackage -Path https://github.com/microsoft/winget-cli/releases/download/v1.5.441-preview/Microsoft.DesktopAppInstaller_8wekyb3d8bbwe.msixbundle` in PowerShell

**Reply by fates7777** [reply]
Add-AppxPackage : Deployment failed with HRESULT: 0x80073CF3, Package failed updates, dependency or conflict 

validation. 

Windows cannot install package Microsoft.WindowsStore_22210.1401.13.0_x64__8wekyb3d8bbwe because this package depends 

on a framework that could not be found. Provide the framework "Microsoft.UI.Xaml.2.7" published by "CN=Microsoft 

Corporation, O=Microsoft Corporation, L=Redmond, S=Washington, C=US", with neutral or x64 processor architecture and 

minimum version 7.2109.13004.0, along with this package to install. The frameworks with name "Microsoft.UI.Xaml.2.7" 

currently installed are: {} 

NOTE: For additional information, look for [ActivityId] 0d81fcb0-45aa-0000-a511-820daa45d901 in the Event Log or use 

the command line Get-AppPackageLog -ActivityID 0d81fcb0-45aa-0000-a511-820daa45d901 

now fixed well thx

**Reply by sigmaphi007** [reply]
> Head over to the [Releases](https://github.com/microsoft/winget-cli/releases) page, go to [1.5.441-preview](https://github.com/microsoft/winget-cli/releases/tag/v1.5.441-preview), download the MSIXBUNDLE, and then open the file. It should prompt to install automatically.
> 
> Alternatively, run `Add-AppxPackage -Path https://github.com/microsoft/winget-cli/releases/download/v1.5.441-preview/Microsoft.DesktopAppInstaller_8wekyb3d8bbwe.msixbundle` in PowerShell

Thank you MSIXBUNDLE worked for me :)

**Reply by davidjenni** [reply]
> Add-AppxPackage : Deployment failed with HRESULT: 0x80073CF3, Package failed updates, dependency or conflict
> validation.
> 
> Windows cannot install package Microsoft.WindowsStore_22210.1401.13.0_x64__8wekyb3d8bbwe because this package depends
> on a framework that could not be found. Provide the framework "Microsoft.UI.Xaml.2.7" published by "CN=Microsoft

To satisfy this MS.UI.Xaml dependency, install it from nuget.org via:
```
$uixaml = Install-Package Microsoft.UI.Xaml -scope currentuser -RequiredVersion 2.7.3 -force
Add-AppxPackage -Path $env:LOCALAPPDATA\PackageManagement\nuget\Packages\Microsoft.UI.Xaml.$($uixaml.Version)\tools\AppX\x64\Release\Microsoft.UI.Xaml.*.appx
Add-AppxPackage -Path https://github.com/microsoft/winget-cli/releases/download/v1.5.441-preview/Microsoft.DesktopAppInstaller_8wekyb3d8bbwe.msixbundle
```

If your Win Powershell doesn't have any nuget package provider setup yet (e.g. on a plain-vanilla Win10):
```
Install-PackageProvider -Name NuGet -Force -Scope CurrentUser
Register-PackageSource -provider NuGet -name nugetRepository -location https://www.nuget.org/api/v2 -Force -Trusted
```
